lib/rpmts.c

```diff
@@ -621,7 +621,11 @@ rpmRC rpmtsImportPubkey(const rpmts ts, const unsigned char * pkt, size_t pktlen
         if (krc != RPMRC_OK) {
             rpmlog(RPMLOG_ERR, "%s\n", lints);
         } else {
-            rpmlog(RPMLOG_WARNING, "%s\n", lints);
+            if ((rpmtsVSFlags(ts) & RPMVSF_NOLINT)) {
+                // Don't show lints when --nolint is provided.
```

Comment:
I think `--nolint` should skip calling pgpPubKeyLint() entirely, that's how such disablers generally behave in rpm. That way it's possible to work around bugs and other gotchas - I only remember regretting *not adding* disablers to some things later, never the other way around.

Reply:
I'm happy to make the change.  But, I just want to be clear what it means.

`pgpPubKeyLint` detects two types of things: hard errors and soft errors (things that look suspicious, but aren't necessarily wrong).  From [my documentation](https://gitlab.com/sequoia-pgp/rpm-sequoia/-/blob/main/src/lib.rs#L1092):

```
/// Lints the first certificate in pkts.
///
/// This function links the certificate according to the current
/// [policy].  It warns about things like unusable subkeys, because they
/// do not have a valid binding signature.  It will also generate a
/// warning if there are no valid, signing-capable keys.
///
/// There are four cases:
///
/// - The packets do not describe a certificate: returns an error and
///   sets `*explanation` to `NULL`.
///
/// - The packets describe a certificate and the certificate is
///   completely unusable: returns an error and sets `*explanation` to
///   a human readable explanation.
///
/// - The packets describe a certificate and some components are not
///   usable: returns success, and sets `*explanation` to a human
///   readable explanation.
///
/// - The packets describe a certificate and there are no lints:
///   returns success, and sets `*explanation` to `NULL`.
```

The change your are proposing would turn off detecting these hard errors.  Please ack that that is really what you want.

Reply:
That'd be in line with what similar other disablers in RPMVSF_* and others do, eg `--noverify`, `--nosignature`, `--nodeps` and all. All very much "apply shotgun to the foot" dangerous by nature if used carelessly and/or without full understanding. But then of course, there are thousands of sanity/structural checks that don't have such disablers. 

I'm looking at this mainly from the "allow working around bugs in (feature X)" but here it's being added for a very different reason actually, and neither seem *quite* right. Maybe I'll need to meditate on this some more :thinking: